reference_guide/custom_language_support/navigation.md

```diff
@@ -0,0 +1,45 @@
+---
+title: Navigation
+---
+<!-- Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file. -->
+
+> **WARNING** This API is available starting from 2020.3 and currently in development and thus in experimental state.
+
+The _Go to Declaration or Usages_ action is performed in several steps.
+
+## Direct navigation
+
+Direct navigation is the navigation from `PsiElement` to another `PsiElement`, 
+such as navigation from `break` keyword to the end of a loop in Java, without showing any popups.
+
+To provide `PsiElement` for direct navigation, implement and register 
+[`DirectNavigationProvider`](upsource:///platform/core-api/src/com/intellij/navigation/DirectNavigationProvider.java).
+
+
+## `Symbol` navigation
+
+If there is no direct navigation available under the caret, then the platform proceeds with `Symbol` navigation.
```

Comment:
```suggestion
If there is no direct navigation available under the caret, then the IntelliJ Platform proceeds with navigation using `Symbol` objects.
```

Reply:
> then the IntelliJ Platform proceeds

Ok

> with navigation using `Symbol` objects.

I'd rather not call symbols _objects_. 

I've used _Symbol navigation_ because this section is called _Symbol navigation_. 
Is _with navigation using symbols_ ok?

Reply:
Apologies, bad suggestion. Saying "with Symbol navigation." is fine, as is "with navigation using symbols."
The itch I was trying to scratch: to me this sentence doesn't clearly define "Symbol navigation", at least not in the sense the 1st sentence under "Direct Navigation" does. Perhaps something like:
"If there is no Direct navigation using a `PsiElement` available under the caret, then the platform proceeds with `Symbol` navigation based on resolving a [reference](declarations_and_references.md#references)." 
(This pulls the link up from the next paragraph.)

Reply:
> this sentence doesn't clearly define "Symbol navigation",

It was not supposed to define Symbol navigation, it's the bit of information that these sections (Direct -> Symbol -> Usages) are performed in order. Initially this sentence was the closing sentence in Direct navigation section, but it's easy to skip this bit if it's in the end of a previous section.

Reply:
I'm not actually sure whether separate definition is needed. I can only come up with "Symbol navigation is navigation based on symbols"

Reply:
I don't want to belabor the point. These are just suggestions. Ultimately, if you are happy with the content then I don't want to delay publication. Go ahead and _resolve conversation_. You have bigger fish to fry.

I do think an introductory paragraph is needed in the Symbol Navigation section to explain what symbol navigation means. Here I have combined the first two paragraphs to make the meaning of Symbol navigation clearer _to me_.
"If there is no Direct navigation available under the caret, then the IntelliJ Platform proceeds with `Symbol` navigation. 
In this method the IntelliJ Platform computes the navigation targets based on resolving a [reference](declarations_and_references.md#references) to obtain the target symbols. 
If there are several target symbols or several navigation targets defined for a symbol, then the IDE shows the navigation popup to ask the user to choose where to go."

Reply:
LGTM, thank you for the suggestion!